src/hipercow/task.py

```diff
@@ -1,7 +1,54 @@
 import pickle
 from dataclasses import dataclass
+from enum import Flag, auto
 
 from hipercow.root import Root
+from hipercow.util import file_create
+
+
+class TaskStatus(Flag):
+    CREATED = auto()
+    SUBMITTED = auto()
+    RUNNING = auto()
+    SUCCESS = auto()
+    FAILURE = auto()
+    CANCELLED = auto()
+    MISSING = auto()
+    TERMINAL = SUCCESS | FAILURE | CANCELLED
+    RUNNABLE = CREATED | SUBMITTED
+
+    def is_runnable(self) -> bool:
+        return bool(self & TaskStatus.RUNNABLE)
+
+    def is_terminal(self) -> bool:
+        return bool(self & TaskStatus.TERMINAL)
+
+    def __str__(self) -> str:
+        return str(self.name).lower()
+
+
+STATUS_FILE_MAP = {
+    TaskStatus.SUCCESS: "status-success",
+    TaskStatus.FAILURE: "status-failure",
+    TaskStatus.CANCELLED: "status-cancelled",
+    TaskStatus.RUNNING: "status-running",
+    TaskStatus.SUBMITTED: "status-submitted",
+}
+
+
+def task_status(root: Root, task_id: str) -> TaskStatus:
+    # check_task_id(task_id)
+    path = root.path / "tasks" / task_id
```

Comment:
I should have thought of this with PR #1 - but do we want to spit task_id into `first two digits / rest` like we did with hipercow R?

Reply:
yup, you're right - I'll do that soon!